src/omlt/base/julia.py

```diff
@@ -0,0 +1,771 @@
+        return OmltExprJuMP(("log", self))
+
+    def tanh(self):
+        return OmltExprJuMP(("tanh", self))
+
+    def __call__(self):
+        """Return the current value of the expression."""
+        return self._jumpexpr.constant + sum(
+            [
+                self._jumpexpr.terms[v] * jump.start_value(v)
+                for v in self._jumpexpr.terms
+            ]
+        )
+
+    def __add__(self, other):
+        return OmltExprJuMP((self, "+", other))
+
+    def __sub__(self, other):
+        return OmltExprJuMP((self, "-", other))
+
+    def __mul__(self, other):
+        return OmltExprJuMP((self, "*", other))
+
+    def __truediv__(self, other):
+        return OmltExprJuMP((self, "/", other))
+
+    def __radd__(self, other):
+        return OmltExprJuMP((self, "+", other))
+
+    def __rmul__(self, other):
+        return OmltExprJuMP((self, "*", other))
+
+    def __rtruediv__(self, other):
+        return OmltExprJuMP((other, "/", self))
+
+    def __eq__(self, other):
+        return OmltConstraintScalarJuMP(lhs=self, sense="==", rhs=other)
+
+    def __le__(self, other):
+        return OmltConstraintScalarJuMP(lhs=self, sense="<=", rhs=other)
+
+    def __ge__(self, other):
+        return OmltConstraintScalarJuMP(lhs=self, sense=">=", rhs=other)
+
+
+# Block
+class OmltBlockJuMP(OmltBlockCore):
+    def __init__(self):
+        self.__formulation = None
+        self.__input_indexes = None
+        self.__output_indexes = None
+        self._format = "jump"
+        self._jumpmodel = jump.Model()
+        self._varrefs = {}
+        self._conrefs = {}
+
+    def __setattr__(self, name, value):
+        super().__setattr__(name, value)
+
+        if isinstance(value, OmltScalarJuMP):
+            value.name = name
+            self._varrefs[name] = jump.add_variable(
+                self._jumpmodel, value.to_jumpvar(), name
+            )
+            value.varref = self._varrefs[name]
+        elif isinstance(value, OmltIndexedJuMP):
+            value.name = name
+            for idx, var in value._vars.items():
+                varname = name + "_" + str(idx)
+                self.__getattribute__(name)[idx].varref = var.add_to_model(
+                    self._jumpmodel, name=varname
+                )
+                self._varrefs[varname] = self.__getattribute__(name)[idx].varref
+
+        elif isinstance(value, OmltConstraintScalarJuMP):
+            self._conrefs[name] = jump.add_constraint(
+                self._jumpmodel, value._jumpcon, name
+            )
+        elif isinstance(value, OmltConstraintIndexedJuMP):
+            value.model = self
+            value.name = name
+        elif isinstance(value, OmltBlockCore):
+            value.name = name
+            value._parent = self
+        elif isinstance(value, dict):
+            for k, v in value.items():
+                v.name = name + "_" + str(k) + "_"
+                v._parent = self
+
+    def set_optimizer(self, optimizer):
```

Comment:
set_optimizer could just be done on the julia side, but no harm in it being here.